### PR TITLE
refactor(datasource/github-releases): Consolidate into single file

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Refactor before converting the datasource to class

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/github-releases/index.ts [hunk 1]
```diff
@@ -1,22 +1,186 @@
...
+          currentVersion: release.tag_name,
+          currentDigest: lineDigest,
+        };
+      }
+    }
+  }
+  return null;
+}
+
+function inferHashAlg(digest: string): string {
```
Comment (viceice) [inline lib/datasource/github-releases/index.ts:44]: Can't we leave those small functions in the common file? So this file don't get so big?

Comment (zharinov) [reply lib/datasource/github-releases/index.ts:44]: Let's do this in the next phase, after migration to class is done

Comment (viceice) [reply lib/datasource/github-releases/index.ts:44]: Why not simply revert this PR changes partially instead, so don't need to change it again?

Comment (zharinov) [reply lib/datasource/github-releases/index.ts:44]: It seems like the initial set of function considered as *util* ones may be different after switching to class-based implementation. I would like to deal with this later, as the separate step.

Comment (viceice) [reply lib/datasource/github-releases/index.ts:44]: Ok

**Review by viceice — APPROVED** [review]

**Pull request merged**